experimental/dds/tree2/api-report/tree2.api.md

```diff
@@ -1228,7 +1243,7 @@ type LazyItem<Item = unknown> = Item | (() => Item);
 export type LazyTreeNodeSchema = TreeNodeSchema | (() => TreeNodeSchema);
 
 // @alpha
-export interface Leaf<TSchema extends LeafSchema> extends TreeNode {
+export interface Leaf<in out TSchema extends LeafSchema> extends TreeNode {
```

Comment:
when working on this change, I think the variance of Sequence changed. It should have always been invariant over its schema, but I think it was covariant before. I have added variance annotations where possible to help ensure these types are invariant (since they are mutable, invariant is what we need).

Note that this isn't robust: the moment these types are anded with another type, these annotations have no effect, but at least putting them here makes the intention clear, and many cases more robust.